Source/Scene/MetadataTableProperty.js

```diff
@@ -316,10 +347,14 @@ function set(property, index, value) {
   }
 
   const classProperty = property._classProperty;
+  const isArray = classProperty.isArray;
+  const type = classProperty.type;
+  const isVectorOrMatrix =
+    MetadataType.isVectorType(type) || MetadataType.isMatrixType(type);
```

Comment:
Should there be a helper for this, called `MetadataType.hasComponents`? If we add other built-in types like quaternions, colors, etc this helper may come in handy.

Reply:
Actually, I recently updated `MetadataType.getComponentCount()` to always return a value even for `STRING` and `ENUM`. checking for `MetadataType.getComponentCount(type) > 1` can be used for this, and will scale better to these cases. 

I'll sweep through and see what parts of the code would benefit from this

Reply:
just updated this. Note that there were a couple places in `MetadataClassProperty` (`isLegacy` and `parseType`) where I intentionally left the `isVector/isMatrix` calls in place to A. be more explicit, and B. because for legacy schemas `type` isn't always a valid `MetadataType` yet